### DISCUSSION TITLE
Work on a strategy of synchronised imports of `motoko-base` for `moc` bumps

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/sdk#2634] [Maintenance Tasks]
We received a report (https://github.com/dfinity/motoko-base/issues/392) that `motoko-base` was too new for the included `moc`.

Suggestion: we could tag (e.g. `moc-0.6.28`) the base library with a _known-to-work_ label, and the `dfx` import procedure could then be modified to always track that for a `moc` bump.

What do you think?

**Comment by viviveevee** [comment]
Sounds useful to me. @ericswanson-dfinity what do you think?

**Comment by crusso** [comment]
It looks to me like dfx post 0.10.1 is just downloading the released moc tar ball from the release site it gets moc from which should reduce the risk.

It would be good to tag motoko-base appropriately too, perhaps just manually until we get add-tags.sh to work again.

I even started on a PR for adapting add-tags.sh but my git/bash foo was too poor
https://github.com/dfinity/motoko/pull/2926

**Comment by ggreif** [comment]
Maybe we can append my `bar` to your `foo` to get some full-fledged `foobar`? :-)

**Comment by crusso** [comment]
Absolutely.

**Comment by ghost** [comment]
We now update `motoko-base` and the `moc` binary based on the version: https://github.com/dfinity/sdk/blob/master/scripts/update-motoko.sh#L13

See also https://github.com/dfinity/sdk/blob/master/nix/sources.json#L159
